spec/util.spec.js

```diff
@@ -1,8 +1,111 @@
 /* eslint-env jasmine */
+/* global ArrayBuffer */
+
 import WaveSurfer from '../src/wavesurfer.js';
 
+/** @test {util.ajax} */
+describe('util.ajax:', function() {
+    var defaultUrl = '/base/spec/support/demo.wav';
+
+    it('can load an arraybuffer', function(done) {
+        var options = {
+            url: defaultUrl,
+            responseType: 'arraybuffer'
+        };
+        var instance = WaveSurfer.util.ajax(options);
+        instance.on('success', (data, e) => {
+            // url
+            expect(e.target.responseURL).toContain(options.url);
+
+            // responseType
+            expect(instance.xhr.responseType).toBe(options.responseType);
+
+            // returned data is an arraybuffer
+            expect(data).toEqual(jasmine.any(ArrayBuffer));
+
+            done();
+        });
+    });
+
+    it('error event is fired when file is not found', function(done) {
+        var options = {
+            url: '/foo/bar'
+        };
+        var instance = WaveSurfer.util.ajax(options);
+        instance.on('error', e => {
+            // url
+            expect(e.target.responseURL).toContain(options.url);
+
+            // error message
+            expect(e.target.statusText).toBe('Not Found');
+            expect(e.target.status).toBe(404);
+
+            done();
+        });
+    });
+
+    it('progress event is fired during loading', function(done) {
+        var options = {
+            url: defaultUrl,
+            responseType: 'arraybuffer'
+        };
+        var instance = WaveSurfer.util.ajax(options);
+        instance.on('progress', e => {
+            // url
+            expect(e.target.responseURL).toContain(options.url);
+
+            // progress message
+            expect(e.target.statusText).toBe('OK');
+            expect(e.target.status).toBe(200);
+
+            done();
+        });
+    });
+
+    it('custom request headers and credentials can be specified', function(done) {
+        var options = {
+            url: defaultUrl,
+            responseType: 'arraybuffer',
+            xhr: {
+                withCredentials: true,
+                requestHeaders: [
+                    {
+                        key: 'Authorization',
+                        value: 'my-token'
+                    }
+                ]
+            }
+        };
+        var instance = WaveSurfer.util.ajax(options);
+        instance.on('success', (data, e) => {
+            // with credentials
+            expect(e.target.withCredentials).toBeTrue();
+
+            // XXX: find a way to retrieve request headers
```

Comment:
@janhartmann any idea for this one? seems setting request headers is possible but i can't find a public property that returns them during this test..